test/modules/paymentProcessor/PP_Connext_Crosschain_v1.t.sol

```diff
@@ -0,0 +1,1131 @@
+error InvalidRecipientOrAmount();
+
+contract PP_Connext_Crosschain_v1_Test is ModuleTest {
+    //--------------------------------------------------------------------------
+    // Constants
+    uint constant MINTED_SUPPLY = 1000 ether;
+    uint constant ZERO_AMOUNT = 0;
+    uint ORIGIN_CHAIN_ID;
+    uint TARGET_CHAIN_ID;
+    //--------------------------------------------------------------------------
+    // Test Storage
+    PP_Connext_Crosschain_v1_Exposed public paymentProcessor;
+    Mock_EverclearPayment public everclearPaymentMock;
+    ERC20PaymentClientBaseV2Mock paymentClient;
+    IPP_Crosschain_v1 public crossChainBase;
+    IWETH public weth;
+
+    // Bridge-related storage
+    address public mockConnextBridge;
+    address public mockEverClearSpoke;
+    address public mockWeth;
+
+    // Execution data storage
+    uint maxFee = 0;
+    uint ttl = 1;
+    bytes32[] public emptyExecutionData;
+
+    //--------------------------------------------------------------------------
+    // Setup Function
+
+    function setUp() public {
+        // Deploy mock contracts and set addresses
+        everclearPaymentMock = new Mock_EverclearPayment();
+        mockEverClearSpoke = address(everclearPaymentMock);
+        mockWeth = address(weth);
+
+        // Deploy payment processor via clone
+        address impl = address(new PP_Connext_Crosschain_v1_Exposed());
+        paymentProcessor = PP_Connext_Crosschain_v1_Exposed(Clones.clone(impl));
+
+        _setUpOrchestrator(paymentProcessor);
+        _authorizer.setIsAuthorized(address(this), true);
+
+        // Initialize payment processor with config
+        bytes memory configData = abi.encode(mockEverClearSpoke, mockWeth);
+        paymentProcessor.init(_orchestrator, _METADATA, configData);
+
+        // Deploy and add payment client through timelock process
+        impl = address(new ERC20PaymentClientBaseV2Mock());
+        paymentClient = ERC20PaymentClientBaseV2Mock(Clones.clone(impl));
+        _orchestrator.initiateAddModuleWithTimelock(address(paymentClient));
+        vm.warp(block.timestamp + _orchestrator.MODULE_UPDATE_TIMELOCK());
+        _orchestrator.executeAddModule(address(paymentClient));
+
+        // Configure payment client
+        paymentClient.init(_orchestrator, _METADATA, bytes(""));
+        paymentClient.setIsAuthorized(address(paymentProcessor), true);
+        paymentClient.setToken(_token);
+
+        ORIGIN_CHAIN_ID = block.chainid;
+        TARGET_CHAIN_ID = 1337;
+
+        emptyExecutionData = new bytes32[](6);
+    }
+
+    //--------------------------------------------------------------------------
+    // Initialization Tests
+
+    /* Test initialization
+    */
+    function testInit() public override(ModuleTest) {
+        assertEq(
+            address(paymentProcessor.orchestrator()), address(_orchestrator)
+        );
+    }
+
+    /* Test interface support
+    */
+    function testSupportsInterface() public {
```

Comment:
Should test for the interface of `IPP_Connext_Crosschain_v1` & `IPaymentProcessor_v1`